tests/units/classes/test.php

```diff
@@ -1245,5 +1250,51 @@ public function testStaticMethod()
 						->isEqualTo(array($return1, $return2, $return3))
 			;
 		}
+
+		public function testNewMockInstance()
+		{
+			$this
+				->if($test = new emptyTest())
+				->then
+					->object($mock = $test->newMockInstance('stdClass'))
+						->isInstanceOf('mock\stdClass')
+						->isInstanceOf('stdClass')
+					->object($test->newMockInstance('stdClass'))
+						->isInstanceOf('mock\stdClass')
+						->isInstanceOf('stdClass')
+						->isNotIdenticalTo($mock)
+					->object($test->newMockInstance('stdClass', 'foobar'))
+						->isInstanceOf('foobar\stdClass')
+						->isInstanceOf('stdClass')
+					->object($test->newMockInstance('stdClass', 'foo', 'bar'))
+						->isInstanceOf('foo\bar')
+						->isInstanceOf('stdClass')
+
+				->given($arguments = array($firstArgument = uniqid(), $secondArgument = rand(0, PHP_INT_MAX)))
+				->then
+					->object($mock = $test->newMockInstance('mageekguy\atoum\dummy', null, null, $arguments))
+						->isInstanceOf('mock\mageekguy\atoum\dummy')
+						->isInstanceOf('mageekguy\atoum\dummy')
+					->mock($mock)
+						->call('__construct')->withArguments($firstArgument, $secondArgument)->once
+
+				->given($arguments = array(uniqid(), rand(0, PHP_INT_MAX), $controller = new mock\controller()))
+				->then
+					->object($mock = $test->newMockInstance('mageekguy\atoum\dummy', null, null, $arguments))
+						->isInstanceOf('mock\mageekguy\atoum\dummy')
+						->isInstanceOf('mageekguy\atoum\dummy')
+					->object($mock->getMockController())->isIdenticalTo($controller)
+
+				->given(
+					$arguments = array(uniqid(), rand(0, PHP_INT_MAX)),
+					$controller = new mock\controller()
```

Comment:
We add an additional argument which is a mock controller and check that atoum correctly injected it into the mock instance (via automagic injection)